src/history/file_backed.rs

```diff
@@ -218,50 +221,34 @@ impl FileBackedHistory {
     }
 
     fn back_with_criteria(&mut self, criteria: &dyn Fn(&str) -> bool) {
-        let mut cursor = self.cursor;
-        let previous_match = self.string_at_cursor();
-
-        while cursor > 0 {
-            cursor -= 1;
-            let entry = &self.entries[cursor];
-            if criteria(entry) {
-                if previous_match
-                    // TODO Get rid of this clone
-                    .clone()
-                    .map_or(false, |value| &value == entry)
-                {
-                    continue;
-                } else {
-                    break;
-                }
+        if !self.entries.is_empty() {
+            if let Some((next_cursor, _)) = self
+                .entries
+                .iter()
+                .take(self.cursor)
+                .enumerate()
+                .rev()
+                .find(|x| criteria(x.1))
```

Comment:
nit-pick: I find `.find(|(_idx, string)| criteria(string)` more readable than this. Thoughts?

Reply:
Will be necessary anyways to fix the consecutive match behavior back in ;)